### PR TITLE
Emulate some mouse movement/press with touch on Android

### PR DESCRIPTION
**Pull request by apazylbe**
Convert touch events to mouse movement:
- First touch will issue left mouse button press
- While the first touch is active it will be tracked as a mouse movement with left button pressed
- Other touches will not be registered
- When the first registered touch is lifted, the tracking will stop, until all touch points have lifted and a new "primary" touch is being tracked

### CODE REVIEW

**Review by Keenuts — COMMENTED** [review]
LGTM for the main bits, just seems like you directly go from WindowImpl to the app, which is not what the GLFW sides does (through WindowEvents, same for the ResizeCallback on Android)

File: include/ppx/application.h [hunk 1]
```diff
@@ -522,6 +526,7 @@ class Application
 
 private:
     friend struct WindowEvents;
+    friend class WindowImplAndroid;
```
[the remark this answers is not in the document]
Comment (apazylbe) [reply include/ppx/application.h]: There was no good reason, changed! 
I had to make functions non static as WindowEvents now has some state (touch id of the first event).

**Review by Keenuts — APPROVED** [review]

File: src/ppx/window_android.cpp [hunk 2]
```diff
@@ -169,7 +177,81 @@ void WindowImplAndroid::OnResizeEvent()
...
     }
     WindowSize size = Size();
-    WindowEvents::ResizeCallback(App(), size.width, size.height);
+    mEvents.ResizeCallback(App(), size.width, size.height);
+}
+
+void WindowEvents::ProcessInputEvent(Application* pApp, AInputEvent* pEvent)
+{
+    int32_t eventType = AInputEvent_getType(pEvent);
+    if (eventType == AINPUT_EVENT_TYPE_MOTION) {
```
[the remark this answers is not in the document]
Comment (apazylbe) [reply src/ppx/window_android.cpp]: fixed, thanks!

File: src/ppx/window_android.cpp [hunk 3]
```diff
@@ -169,7 +177,81 @@ void WindowImplAndroid::OnResizeEvent()
...
+            case AMOTION_EVENT_ACTION_MOVE: {
+                if (mFirstTouchId != -1) {
+                    // Only track the first registered touch event.
+                    float x = AMotionEvent_getX(pEvent, 0);
+                    float y = AMotionEvent_getY(pEvent, 0);
+                    pApp->MouseMoveCallback(x, y, MOUSE_BUTTON_LEFT);
+                }
+                break;
+            }
+            default:
```
[the remark this answers is not in the document]
Comment (apazylbe) [reply src/ppx/window_android.cpp]: fixed, thanks!

**Pull request merged**